mr/src/main/java/org/apache/iceberg/mr/mapred/TableResolver.java

```diff
@@ -45,27 +45,26 @@ static Table resolveTableFromJob(JobConf conf) throws IOException {
 
   static Table resolveTableFromConfiguration(Configuration conf, Properties properties) throws IOException {
     String catalogName = properties.getProperty(InputFormatConfig.CATALOG_NAME, InputFormatConfig.HADOOP_TABLES);
-    String tableLocation = properties.getProperty(InputFormatConfig.TABLE_LOCATION);
-    String tableName = properties.getProperty(InputFormatConfig.TABLE_NAME);
-    Preconditions.checkNotNull(tableLocation, "Table location is not set.");
-    Preconditions.checkNotNull(tableName, "Table name is not set.");
+
     switch (catalogName) {
       case InputFormatConfig.HADOOP_TABLES:
+        String tableLocation = properties.getProperty(InputFormatConfig.TABLE_LOCATION);
+        Preconditions.checkNotNull(tableLocation, "Table location is not set.");
         HadoopTables tables = new HadoopTables(conf);
         return tables.load(tableLocation);
+
       case InputFormatConfig.HIVE_CATALOG:
+        String tableName = properties.getProperty(InputFormatConfig.TABLE_NAME);
+        Preconditions.checkNotNull(tableName, "Table name is not set.");
         //TODO Implement HiveCatalog
         return null;
       default:
-        throw new NoSuchTableException("Table does not exist at location: " + tableLocation);
+        throw new RuntimeException("Catalog " + catalogName + " not supported.");
```

Comment:
nit: throw `NoSuchNamespacException`

Reply:
Sure, we can address that in https://github.com/apache/iceberg/pull/1103/files#diff-64ac74e3513bf6536c0a39f377c48ce5R62 (I'll make the change now, also spotted that the tests for this class could be fleshed out and tidied up a bit)